subsys/bluetooth/mesh/cfg_srv.c

```diff
@@ -75,27 +79,66 @@ static int dev_comp_data_get(struct bt_mesh_model *model,
 	bt_mesh_model_msg_init(&sdu, OP_DEV_COMP_DATA_STATUS);
 
 	net_buf_simple_add_u8(&sdu, page);
-	if (IS_ENABLED(CONFIG_BT_MESH_COMP_PAGE_1) && page == 1) {
-		err = bt_mesh_comp_data_get_page_1(&sdu);
+
+	switch (page) {
+	case 0:
+		if (atomic_test_bit(bt_mesh.flags, BT_MESH_COMP_DIRTY)) {
+			sdu.size -= BT_MESH_MIC_SHORT;
+			err = bt_mesh_comp_read(&sdu, 0);
+			if (err) {
+				LOG_ERR("Unable to get stored composition data");
+				return err;
+			}
+
+			sdu.size += BT_MESH_MIC_SHORT;
+		} else {
+			err = bt_mesh_comp_data_get_page_0(&sdu, 0);
+			if (err < 0) {
+				LOG_ERR("Unable to get composition page 0");
+				return err;
+			}
+		}
+		break;
+
+	case 1:
+		if (atomic_test_bit(bt_mesh.flags, BT_MESH_COMP_DIRTY)) {
+			sdu.size -= BT_MESH_MIC_SHORT;
+			err = bt_mesh_comp_read(&sdu, 1);
+			if (err) {
+				LOG_ERR("Unable to get stored composition data");
+				return err;
+			}
+
+			sdu.size += BT_MESH_MIC_SHORT;
+		} else {
+			err = bt_mesh_comp_data_get_page_1(&sdu);
+		}
 		if (err < 0) {
 			LOG_ERR("Unable to get composition page 1");
 			return err;
 		}
-	} else if (atomic_test_bit(bt_mesh.flags, BT_MESH_COMP_DIRTY) && (page == 0U)) {
-		sdu.size -= BT_MESH_MIC_SHORT;
-		err = bt_mesh_comp_read(&sdu);
-		if (err) {
-			LOG_ERR("Unable to get stored composition data");
-			return err;
-		}
+		break;
 
-		sdu.size += BT_MESH_MIC_SHORT;
-	} else {
+	case 128:
 		err = bt_mesh_comp_data_get_page_0(&sdu, 0);
 		if (err < 0) {
 			LOG_ERR("Unable to get composition page 0");
 			return err;
 		}
+		break;
+
+	case 129:
+		err = bt_mesh_comp_data_get_page_1(&sdu);
+		if (err < 0) {
+			LOG_ERR("Unable to get composition page 1");
+			return err;
+		}
+		break;
+
+	default:
+		/*Should never happen*/
```

Comment:
Why should it never happen? What if a client requested a page that the server doesn't support? What server should reply?

Reply:
Discussed in person